testing/test_session.py

```diff
@@ -253,3 +253,28 @@ def pytest_sessionfinish():
     """)
     res = testdir.runpytest("--collect-only")
     assert res.ret == EXIT_NOTESTSCOLLECTED
+
+
+def test_rootdir_option_arg(testdir):
```

Comment:
Please also add tests for absolute and paths containing environment variables.

Reply:
paths containing environment variables not working in tests, but working when I test locally :(

```
testdir.runpytest("--rootdir=$HOME")

ERROR: Directory '/tmp/pytest-of-feuillemorte/pytest-31/test_rootdir_option_arg2/$HOME/root' not found. Check your '--rootdir' option.
```

Reply:
You need to explicitly call `os.path.expandvars` to deal with variable expansion.

When you say "locally" I suppose you mean the command line? In that case is your shell doing the expansion, not pytest. 😉